From 47b2889da270f1f54ffb8a903ed16a65d38091c2 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Mon, 24 Jun 2019 12:58:49 +0530
Subject: [PATCH 089/138] octeontx2-af: Display message timeout error

commit 8b24f72eca1448c7836349bba0bc2126722b9b5f from
git@git.assembla.com:cavium/WindRiver.linux.git

Displaying an error if time out occurs while waiting
for response helps in debugging.

Change-Id: I9532501d1cc5e60e9364f94f20d79b585ea03204
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/11504
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/mbox.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.c b/drivers/net/ethernet/marvell/octeontx2/af/mbox.c
index 547d473fad8d..75dbcacc23c9 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.c
@@ -137,13 +137,16 @@ EXPORT_SYMBOL(otx2_mbox_init);
 int otx2_mbox_wait_for_rsp(struct otx2_mbox *mbox, int devid)
 {
 	struct otx2_mbox_dev *mdev = &mbox->dev[devid];
+	struct device *sender = &mbox->pdev->dev;
 	int timeout = 0, sleep = 1;
 
 	while (mdev->num_msgs != mdev->msgs_acked) {
 		msleep(sleep);
 		timeout += sleep;
-		if (timeout >= MBOX_RSP_TIMEOUT)
+		if (timeout >= MBOX_RSP_TIMEOUT) {
+			dev_err(sender, "timed out while waiting for rsp\n");
 			return -EIO;
+		}
 	}
 	return 0;
 }
-- 
2.17.1

